api/src/main/java/com/google/appengine/api/datastore/IndexTranslator.java

```diff
@@ -73,6 +74,7 @@ public static Index.Property convertFromPb(OnestoreEntity.Index.Property propert
   }
 
   public static Index convertFromPb(OnestoreEntity.Index index) {
-    return convertFromPb(OnestoreEntity.CompositeIndex.newBuilder().setId(0).setDefinition(index).build());
+    return convertFromPb(OnestoreEntity.CompositeIndex.newBuilder().setId(0).setDefinition(index).setAppId("").setState(
```

Comment:
Hum, sure an emtpy appId is fine there? Maybe, but not sure!

Reply:
I'm using the values that are present in the private default constructor